### PR TITLE
Flag empty list concatenation

### PR DESCRIPTION
**Pull request by viraptor**
`[] ++ something` should be reduced to just `something`

### CODE REVIEW

File: lib/src/lints/empty_list_concat.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+use if_chain::if_chain;
+use macros::lint;
+use rnix::{
+    types::{BinOp, BinOpKind, TypedNode, List},
+    NodeOrToken, SyntaxElement, SyntaxKind, SyntaxNode,
+};
+
+/// ## What it does
+/// Checks for empty list being concatenated.
```
Comment (nerdypepper) [inline lib/src/lints/empty_list_concat.rs:11]:
```suggestion
/// Checks for concatenations to empty lists
```

File: lib/src/lints/empty_list_concat.rs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+fn is_empty_array(node: &SyntaxNode) -> bool {
+    if_chain! {
+        if let Some(list) = List::cast(node.clone());
+        then {
+            list.items().count() == 0
+        } else {
+            false
+        }
+    }
```
Comment (nerdypepper) [inline lib/src/lints/empty_list_concat.rs:68]: can we omit the use of the `if_chain` macro here?

**Review by nerdypepper — CHANGES_REQUESTED** [review]
Thanks for the PR! Apart from minor code changes, there is one more addition: unit tests, please add snapshot data to `bin/tests/data` and run `cargo insta review`

**Comment by viraptor** [comment]
Done, tests included.

**Review by ilkecan — APPROVED** [review]

**Pull request merged**